gettsim/interface.py

```diff
@@ -335,13 +335,13 @@ def _fail_if_datatype_is_false(data, columns_overriding_functions, functions):
         check_data = True
         if column_name in TYPES_INPUT_VARIABLES:
             internal_type = TYPES_INPUT_VARIABLES[column_name]
-            check_data = check_if_series_has_internal_type(series, internal_type)
+            check_data, _ = convert_series_to_internal_type(series, internal_type)
         elif (
             column_name in columns_overriding_functions
             and "return" in functions[column_name].__annotations__
         ):
             internal_type = functions[column_name].__annotations__["return"]
-            check_data = check_if_series_has_internal_type(series, internal_type)
+            check_data, _ = convert_series_to_internal_type(series, internal_type)
 
         if not check_data:
```

Comment:
The variable check_data and the error below shouldn't be necessary anymore as the function should handle potential problems internally. (see the other comments)

Reply:
Can't this function be fully deleted now?

Reply:
We still need a function which converts all input data to the expected type.

I think we can still use the structure of this function, but replace the two lines by:

`data[column_name] = convert_series_to_internal_type(series, internal_type)`